services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java

```diff
@@ -560,8 +562,10 @@ public Stream<UserSessionRepresentation> getOfflineUserSessions(@QueryParam("fir
         firstResult = firstResult != null ? firstResult : -1;
         maxResults = maxResults != null ? maxResults : Constants.DEFAULT_MAX_RESULTS;
 
-        return session.sessions().getOfflineUserSessionsStream(client.getRealm(), client, firstResult, maxResults)
-                .map(this::toUserSessionRepresentation);
+        return session.sessions()
+                .getOfflineUserSessionsStream(client.getRealm(), client, firstResult, maxResults)
+                .map(this::toUserSessionRepresentation)
```

Comment:
It should be safer to filter sessions in `InfinispanUserSessionProvider` and avoid changing its callers.

Reply:
Same comment as above https://github.com/keycloak/keycloak/pull/17577#discussion_r1162283337